examples/form_data/form_data.rs

```diff
@@ -14,15 +13,16 @@ fn main() {
     });
 
     server.post("/confirmation", middleware!{ |req, res|
-        let mut form_data = String::new();
-        req.origin.read_to_string(&mut form_data).unwrap();
+        let form_data = try_with!(res, req.post());
 
-        println!("{}", form_data);
+        println!("{:?}", form_data);
 
         let mut data = HashMap::new();
         data.insert("title", "Confirmation");
-        data.insert("formData", &form_data);
-
+        data.insert("firstname", form_data.get("firstname").unwrap_or("What's your first name?"));
```

Comment:
Also, it would be good to see what matching on the error looks like, I know you raised the point about returning `(StatusCode, Error)` in the other thread. Perhaps it's better to introduce a new error type? 

How should things be handled if the method is called when no form-data is available, or if it was posted as json/<other-data-format> instead?